Assets/_Project/Code/Scripts/QuestionStage/AnswerControl.cs

```diff
@@ -0,0 +1,238 @@
+﻿using System;
+using System.Collections;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+using Unity.Netcode;
+using System.Linq;
+public class AnswerControl : NetworkBehaviour
+{
+    public TMP_InputField answerInput;
+    public TMP_Text questionText;
+    public TMP_Text feedbackText;
+    public TMP_Text timerText;
+    public Button submitButton;
+    public TMP_Text numerRundy;
+    public TMP_Text buttonText1, buttonText2, buttonText3, buttonText4;
+    public Button answerButton1, answerButton2, answerButton3, answerButton4;
+    public Button hintButton;
+
+    public static ulong IdOfPlayerWhoWonAuction = 0;
+    public static int currentQuestionIndex = 0;
+    private float _timeRemaining;
+    private const int _totalQuestions = 8;
+    private bool _isAnswerChecked;
+
+    public static Category category;
+    public Question currentQuestion;
+
+    void Start()
+    {
+        interactivityOfItems(false);
+        currentQuestionIndex++;
+        _isAnswerChecked = false;
+        HintMode(false);
+
+        answerButton1.onClick.AddListener(() => SelectButton(answerButton1));
+        answerButton2.onClick.AddListener(() => SelectButton(answerButton2));
+        answerButton3.onClick.AddListener(() => SelectButton(answerButton3));
+        answerButton4.onClick.AddListener(() => SelectButton(answerButton4));
+
+        numerRundy.text = "Runda numer: " + currentQuestionIndex.ToString();
+        if (IsHost)
+        {
+            SendCategoryToServerRpc();
+        }
+        if (NetworkManager.Singleton.LocalClientId == IdOfPlayerWhoWonAuction)
+        {
+            feedbackText.text = "Jesteś graczem ktory wygrał licytacje";
+        }
+        else
+        {
+            feedbackText.text = "Jesteś graczem ktory przegrał licytacje. Tryb obserwatora";
+        }
+    }
+
+    private IEnumerator StartCountdown()
+    {
+        while (_timeRemaining > 0 && _isAnswerChecked == false)
+        {
+            _timeRemaining -= Time.deltaTime;
+            showCurrentTimeRpc(_timeRemaining);
+            yield return null;
+        }
+        if (_timeRemaining <= 0)
+        {
+            interactivityOfItems(false);
+            feedbackText.text = "Czas minął! Odpowiedź: " + currentQuestion.giveCorrectAnswer();
+            StartCoroutine(ChangeScene("Lobby", 4));
+        }
+    }
+
+    [Rpc(SendTo.Server)]
+    void timeControllerRpc()
```

Comment:
tutaj też, to jest dobra nazwa pola, ale jako metoda nie określa co ta metoda robi, StartServerTimerRpc() np. żeby zprecyzować że timer odpalany jest na serwerze